### PR TITLE
Config test suites

### LINKED ISSUE #555: Allowlist Config test suites
We can add allowlist config test suites similar to what we did with #497 for contract tests.

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Adds test suites for config tests

## How this works

Similar to Contract tests, it adds a AllowList test suite for `Equal` and `Verify` tests

## How this was tested

Added tests that test those tests.

## How is this documented

Not yet documented, will document together after #556

Closes #555

### CODE REVIEW

File: precompile/allowlist/config_test.go [hunk 1]
```diff
@@ -6,92 +6,23 @@ package allowlist
...
-		})
+	dummyModule := modules.Module{
+		Address:      dummyAddr,
+		Contract:     CreateAllowListPrecompile(dummyAddr),
+		Configurator: &dummyConfigurator{},
 	}
+	VerifyPrecompileWithAllowListTests(t, dummyModule, nil)
 }
 
 func TestEqualAllowListAllowList(t *testing.T) {
```
Comment (darioush) [inline precompile/allowlist/config_test.go:21]: should we rename this test to `TestEqualAllowList`?

File: precompile/contracts/feemanager/config_test.go [hunk 2]
```diff
@@ -27,106 +28,60 @@ var validFeeConfig = commontype.FeeConfig{
...
-			config:        NewConfig(big.NewInt(3), admins, nil, &invalidFeeConfig),
+	tests := map[string]testutils.ConfigVerifyTest{
+		"invalid initial fee manager config": {
+			Config:        NewConfig(big.NewInt(3), admins, nil, &invalidFeeConfig),
 			ExpectedError: "gasLimit = 0 cannot be less than or equal to 0",
-		},
-		{
-			name:          "nil initial fee manager config",
-			config:        NewConfig(big.NewInt(3), admins, nil, &commontype.FeeConfig{}),
+		}, "nil initial fee manager config": {
```
Comment (darioush) [inline precompile/contracts/feemanager/config_test.go:38]: can we add a line break after `,`?

**Review by darioush — APPROVED** [review]

File: precompile/contracts/feemanager/config_test.go (hunk 2 above)
Comment (ceyonur) [reply precompile/contracts/feemanager/config_test.go]: added

File: precompile/contracts/rewardmanager/config_test.go [hunk 3]
```diff
@@ -7,115 +7,77 @@ import (
...
 		},
-		{
-			name:     "different enabled",
-			config:   NewConfig(big.NewInt(3), admins, nil, nil),
-			other:    NewConfig(big.NewInt(3), admins, enableds, nil),
-			expected: false,
+		"different enabled": {
+			Config:   NewConfig(big.NewInt(3), admins, nil, nil),
+			Other:    NewConfig(big.NewInt(3), admins, enableds, nil),
+			Expected: false,
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply precompile/contracts/rewardmanager/config_test.go]: yep correct, removed.

File: precompile/allowlist/config_test.go [hunk 4]
```diff
@@ -6,92 +6,23 @@ package allowlist
...
-			require := require.New(t)
-
-			err := tt.config.Verify()
-			if tt.expectedError == "" {
-				require.NoError(err)
-			} else {
-				require.ErrorContains(err, tt.expectedError)
-			}
-		})
+	dummyModule := modules.Module{
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply precompile/allowlist/config_test.go]: moved

[the remark this answers is not in the document]
Comment (ceyonur) [reply precompile/allowlist/config_test.go]: chanegd

**Pull request merged**